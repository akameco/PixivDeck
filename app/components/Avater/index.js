// @flow
import styled from 'styled-components'
import React from 'react'

type Props = {
  size: number,
  src: string,
  style?: Object,
  className?: string,
  color?: string,
}

const StyledAvater = styled.img`
	user-select: none;
	display: inline-flex;
	align-items: center;
	justify-content: center;
	border-radius: 5px;
	width: ${props => props.size}px;
	height: ${props => props.size}px;
	color: ${props => props.color || '#fff'};
`

const Avater = (props: Props) => <StyledAvater {...props} />

export default Avater
