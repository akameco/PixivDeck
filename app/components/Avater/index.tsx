import styled from 'styled-components'
import * as React from 'react'

interface Props {
  size: number
  src: string
  style?: object
  className?: string
  color?: string
}
const StyledAvater = styled.img`
  user-select: none;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 5px;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  color: ${props => props.color || '#fff'};
`

const Avater = (props: Props) => <StyledAvater {...props} />

export default Avater
