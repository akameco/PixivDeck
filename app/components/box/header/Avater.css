.avater {
	width: 48px;
	height: 48px;
	margin: 5px;
}

.avater img {
	width: 50px;
	height: 50px;
	border-radius: 4px;
}
