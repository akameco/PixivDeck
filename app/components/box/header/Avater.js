// @flow
import React from 'react'
import styles from './Avater.css'

type Props = {
	img: string
};

const Avater = ({img}: Props) => (
	<div className={styles.avater}>
		<img src={img}/>
	</div>
)

export default Avater
