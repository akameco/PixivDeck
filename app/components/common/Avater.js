// @flow
import React from 'react'

type Props = {
	size: number,
	src: string,
	style?: Object,
	className?: string,
	color?: string,
}

const getStyles = ({color, size}: $Shape<Props>): Object => {
	const styles = {
		root: {
			userSelect: 'none',
			display: 'inline-flex',
			alignItems: 'center',
			justifyContent: 'center',
			fontSize: size / 2,
			borderRadius: '5px',
			height: size,
			width: size,
			color: color || '#fff',
		},
	}
	return styles
}

const Avater = ({src, style, className, ...others}: Props) => (
	<img
		src={src}
		style={Object.assign(getStyles(others).root, style)}
		className={className}
		/>
)

export default Avater
