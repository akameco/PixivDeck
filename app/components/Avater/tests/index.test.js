import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import Avater from '../'

const renderComponent = (props = {}) => shallow(
	<Avater {...props}/>
)

test('snapshot', () => {
	const c = renderComponent()
	expect(shallowToJson(c)).toMatchSnapshot()
})
