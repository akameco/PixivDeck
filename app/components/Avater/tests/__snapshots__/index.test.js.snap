exports[`test snapshot 1`] = `<styled.img />`;
